flink-runtime/src/main/java/org/apache/flink/runtime/codegeneration/SorterTemplateModel.java

```diff
@@ -0,0 +1,342 @@
+	private final static HashMap<Integer,String> byteOperatorMapping = new HashMap<Integer, String>(){
+		{
+			put(8, "Long");
+			put(4, "Int");
+			put(2, "Short");
+			put(1, "Byte");
+		}
+	};
+
+	// ------------------------------------------------------------------------
+	//                                   Attributes
+	// ------------------------------------------------------------------------
+	private final TypeComparator typeComparator;
+	private final ArrayList<Integer> fixedByteChunks;
+	private final String sorterName;
+
+	// no. bytes of a sorting key
+	private final int numKeyBytes;
+
+	// used to determine whether order of records can completely determined by normalized sorting key
+	// or the sorter has to also deserialize records if their keys are equal to really confirm the order
+	private final boolean isKeyFullyDetermined;
+
+	/**
+	 * Constructor
+	 * @param typeComparator
+	 * 		  The type information of underlying data
+	 */
+	public SorterTemplateModel(TypeComparator typeComparator){
+		this.typeComparator = typeComparator;
+
+		// compute no. bytes for sorting records and check whether this bytes is just prefix or not.
+		if (this.typeComparator.supportsNormalizedKey()) {
+			// compute the max normalized key length
+			int numPartialKeys;
+			try {
+				numPartialKeys = this.typeComparator.getFlatComparators().length;
+			} catch (Throwable t) {
+				numPartialKeys = 1;
+			}
+
+			int maxLen = Math.min( NormalizedKeySorter.DEFAULT_MAX_NORMALIZED_KEY_LEN, NormalizedKeySorter.MAX_NORMALIZED_KEY_LEN_PER_ELEMENT * numPartialKeys);
+
+			this.numKeyBytes = Math.min(this.typeComparator.getNormalizeKeyLen(), maxLen);
+			this.isKeyFullyDetermined = !this.typeComparator.isNormalizedKeyPrefixOnly(this.numKeyBytes);
+		}
+		else {
+			this.numKeyBytes = 0;
+			this.isKeyFullyDetermined = false;
+		}
+
+		// split key into fixed-byte chunks
+		this.fixedByteChunks = generatedSequenceFixedByteChunks(this.numKeyBytes);
+
+		this.sorterName   = generateCodeFilename(this.fixedByteChunks, this.isKeyFullyDetermined);
+	}
+
+	// ------------------------------------------------------------------------
+	//                               Public Methods
+	// ------------------------------------------------------------------------
+
+	public Map<String,String> getTemplateVariables() {
+
+		Map<String,String> templateVariables = new HashMap();
+
+		templateVariables.put("name", this.sorterName);
+
+		String swapProcedures  	 = generateSwapProcedures();
+		String writeProcedures   = generateWriteProcedures();
+		String compareProcedures = generateCompareProcedures();
+
+		templateVariables.put("writeProcedures", writeProcedures);
+		templateVariables.put("swapProcedures", swapProcedures);
+		templateVariables.put("compareProcedures", compareProcedures);
+
+
+		return templateVariables;
+	}
+
+	/**
+	 * Getter for sorterName which instantiated from constructor
+	 * @return name of the sorter
+	 */
+	public String getSorterName (){
```

Comment:
It's just that there is a space before `()` but not after, and in other methods the space is after `()`.